Sources/DeepLearning/Operators/Basic.swift

```diff
@@ -0,0 +1,350 @@
+    ///
+    /// The coordinates are returned in a 2-D tensor where the first dimension (rows) represents the 
+    /// number of non-zero elements, and the second dimension (columns) represents the coordinates 
+    /// of the non-zero elements. Keep in mind that the shape of the output tensor can vary 
+    /// depending on how many true values there are in this tensor. Indices are output in row-major 
+    /// order.
+    ///
+    /// For example:
+    /// ```
+    /// // 'input' is [[true, false], [true, false]]
+    /// // 'input' has 2 true values and so the output has 2 rows.
+    /// // 'input' has rank of 2, and so the second dimension of the output has size 2.
+    /// input.nonZeroIndices() // is [[0, 0], [1, 0]]
+    ///
+    /// // 'input' is [[[ true, false], [ true, false]],
+    /// //             [[false,  true], [false,  true]],
+    /// //             [[false, false], [false,  true]]]
+    /// // 'input' has 5 true values and so the output has 5 rows.
+    /// // 'input' has rank 3, and so the second dimension of the output has size 3.
+    /// input.nonZeroIndices() // is [[0, 0, 0],
+    ///                        //     [0, 1, 0],
+    ///                        //     [1, 0, 1],
+    ///                        //     [1, 1, 1],
+    ///                        //     [2, 1, 1]]
+    /// ```
+    ///
+    /// - Returns: A tensor with shape `(num_true, rank(condition))`.
+    @inlinable
+    func nonZeroIndices() -> Tensor<Int64> {
+        return Raw.where_(self)
+    }
+}
+
+public extension Tensor where Scalar == Bool {
+    /// Returns the elements of either `x` or `y`, depending on the values in stored in this tensor.
+    /// 
+    /// `x` and `y` must be scalar if this tensor is scalar. Otherwise, either the first dimension 
+    /// of `x` and `y` must match the shape of this tensor (i.e., this tensor must be a vector), or 
+    /// the shapes of `x` and `y` must match the shape of this tensor. This tensor acts as a mask 
+    /// that chooses, based on the value at each element, whether the corresponding element / row in 
+    /// the output should be taken from `x` (if true) or `y` (if false). If this tensor is a vector 
+    /// and `x` and `y` are higher rank matrices, then it chooses which row (outer dimension) to 
+    /// copy from `x` and `y`. If it has the same shape as `x` and `y`, then it chooses which 
+    /// element to copy from `x` and `y`.
+    /// 
+    /// - Parameters:
+    ///   - x: Contains the values to use when the condition is true.
+    ///   - y: Contains the values to use when the condition is false.
+    /// 
+    /// - Precondition: `x` and `y` must have the same shape.
+    /// 
+    /// - Returns: A tensor with the same type and shape as `x` and `y`.
+    @inlinable
+    @differentiable(
+        wrt: (x, y),
+        vjp: _vjpSelecting(ifTrue:else:) where T: TensorFlowFloatingPoint)
+    func selecting<T: TensorFlowScalar>(
```

Comment:
We already have this op. It's called [`replacing(with:where:)`](https://github.com/apple/swift/blob/9340e362563b19eac7a1708c176c97ae900dfa2f/stdlib/public/TensorFlow/Ops.swift#L1006) following the Swift [SIMD vectors](https://github.com/apple/swift-evolution/blob/master/proposals/0229-simd.md) proposal.